docs/layout-of-source-files.rst

```diff
@@ -26,8 +26,7 @@ The version pragma is used as follows::
   pragma solidity ^0.4.0;
 
 Such a source file will not compile with a compiler earlier than version 0.4.0
-and it will also not work on a compiler starting from version 0.5.0 (this
-second condition is added by using ``^``). The idea behind this is that
+and it will also not work on a compiler starting from version 0.5.0. The idea behind this is that
```

Comment:
Why is this text better?

Reply:
@axic This text is better because `0.4.0` **without the carrot** will also `not work on a compiler starting from version 0.5.0`.  The above is opening a can of worms.  This document should try to avoid duplicating documentation that better stated else where such as: https://docs.npmjs.com/misc/semver

Reply:
While I agree, I still believe that we should at least explain a bit about what the reason behind this very common pattern is. Only the very interested will read the npm semver docs, so we should give some kind of tldr.

Would it be a compromise to add the link already here?

Reply:
I'd vote to have a short intro to semver with some useful patterns we use (`^0.4.24`, `>0.4.24 <0.5.0`, `0.4.24`).

Reply:
Agreed!